skimage/typing/_image_types.py

```diff
@@ -0,0 +1,13 @@
+import numpy as np
+from typing_extensions import Annotated
+
+Image3D = Annotated[np.ndarray, 'image_3d',
+                        'a 2D or 3D single or multichannel image']
+Image2D = Annotated[np.ndarray, 'image_2d',
+                        'a 2D single or multichannel image']
```

Comment:
Maybe I'm missing something but shouldn't `Image2D` and `Image3D` be subsets of `Image`, e.g. `Image2D` is a valid value for `Image` but not the other way round.

Reply:
@lagru at the moment we have no way to define a type object which would match only arrays of dimension 2 or 3. So what counts in the annotations above is the object `np.ndarray` (which can be changed later on to accept duck arrays), the rest is sugar which can possibly be used at runtime but not by mypy.

Reply:
Ah, that makes sense. Thank you for explaining.